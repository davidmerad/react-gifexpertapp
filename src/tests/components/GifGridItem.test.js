import { shallow } from "enzyme"
import { GifGridItem } from "../../components/GifGridItem"

describe('Pruebas en <GifGridItem />', () => {
    let title = 'titulo';
    let url = 'https://localhost/algo.jps';

    let wrapper = shallow(<GifGridItem title={title} url={url} />);
    beforeEach(() => {
        wrapper = shallow(<GifGridItem title={title} url={url} />); 
    });

    test('Debe mostrar <GifGridItem /> correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
    test('Debe tener un parrafo con el texto', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    });
    
    test('Debe tener una imagen igual al URL y alt de los props', () => {
       const img = wrapper.find('img');
       const urlt = img.props().src;
       const alt = img.props().alt;
       expect(alt.trim()).toBe(title);
       expect(urlt).toBe(url); 
    });
    
    test('Debe tener animate__flipInY', () => {
        const img = wrapper.find('div');
        
        expect(img.hasClass('animate__flipInY')).toBe(true);
    });
    

})
