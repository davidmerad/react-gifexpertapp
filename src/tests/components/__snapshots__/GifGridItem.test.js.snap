// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Debe mostrar <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__flipInY"
>
  <img
    alt="titulo"
    src="https://localhost/algo.jps"
  />
  <p>
    titulo
  </p>
</div>
`;
